server/pulp/server/webservices/views/consumers.py

```diff
@@ -0,0 +1,1082 @@
+        :raises InvalidValue: if some parameters are invalid
+
+        :return: Response representing the deleted binding(in case 'notify agent' is set to False)
+        :rtype: django.http.HttpResponse
+        """
+
+        body = request.body_as_json
+        forced = body.get('force', False)
+        if not isinstance(forced, bool):
+            raise InvalidValue(['force'])
+        options = body.get('options', {})
+        if not isinstance(options, dict):
+            raise InvalidValue(['options'])
+        if forced:
+            call_report = consumer_task.force_unbind(consumer_id, repo_id, distributor_id, options)
+        else:
+            call_report = consumer_task.unbind(consumer_id, repo_id, distributor_id, options)
+
+        if call_report.spawned_tasks:
+            raise OperationPostponed(call_report)
+        else:
+            return generate_json_response_with_pulp_encoder(call_report.serialize())
+
+
+class ConsumerContentActionView(View):
+    """
+    Views for content manipulation on the consumer.
+    """
+
+    @auth_required(authorization.CREATE)
+    @json_body_required
+    def post(self, request, consumer_id, action):
+        """
+        Install/update/uninstall content unit/s on the consumer.
+
+        :param request: WSGI request object
+        :type request: django.core.handlers.wsgi.WSGIRequest
+        :param consumer_id: A consumer ID.
+        :type consumer_id: str
+        :param action: type of action to perform
+        :type action: str
+
+        :raises MissingResource: if consumer id does not exist
+        """
+
+        method = getattr(self, action, None)
+        if method:
+            try:
+                factory.consumer_manager().get_consumer(consumer_id)
+            except:
+                raise MissingResource(consumer_id=consumer_id)
+            return method(request, consumer_id)
+        else:
+            return HttpResponseBadRequest('bad request')
+
+    def install(self, request, consumer_id):
+        """
+        Install content (units) on a consumer.
+
+        Expected body: {units:[], options:<dict>}
+        where unit is: {type_id:<str>, unit_key={}} and the
+        options is a dict of install options.
+
+        :param request: WSGI request object
+        :type request: django.core.handlers.wsgi.WSGIRequest
+        :param consumer_id: A consumer ID.
+        :type consumer_id: str
+
+        :raises MissingValue: if some values are missing
+        :raises OperationPostponed: when an async operation is performed.
+        """
+
+        body = request.body_as_json
+        missing_params = []
+        units = body.get('units')
+        if units is None:
+            missing_params.append('units')
+        options = body.get('options')
+        if options is None:
+            missing_params.append('options')
+
+        if missing_params:
+            raise MissingValue(missing_params)
```

Comment:
Looks like options and units are required for install, update, and uninstall. I think you could move this up to `post` and remove some code and several redundant tests.